services/src/main/java/io/druid/cli/CliInternalHadoopIndexer.java

```diff
@@ -150,7 +144,7 @@ public HadoopDruidIndexerConfig getHadoopDruidIndexerConfig()
             final URI argumentSpecUri = new URI(argumentSpec);
             final String argumentSpecScheme = argumentSpecUri.getScheme();
 
-            if (argumentSpecScheme == null || argumentSpecScheme.equals("file")) {
+            if (argumentSpecScheme == null || "file".equals(argumentSpecScheme)) {
```

Comment:
Also worth catching this consistently, via IntelliJ inspection

Reply:
Yes, it is.

Reply:
Original intellij inspection profile only catches `obj != null && obj.equals("")` situation, rather than `obj.equals("")`

![image](https://user-images.githubusercontent.com/8108788/42548521-fda35870-84f9-11e8-9846-fdfa0ac34d9d.png)

After adding `AliEqualsAvoidNull` inspection into profile, it can recognizes `obj.equals("")` case

![image](https://user-images.githubusercontent.com/8108788/42548595-6f5e7ff8-84fa-11e8-89c9-48cef546c667.png)

But, `AliEqualsAvoidNull` inspection need [P3C](https://github.com/alibaba/p3c) plugin... What you think? @leventov

Reply:
PS: The origin intellij inspection about string equals is `SimplifiableEqualsExpression`

Reply:
I think you could catch those things merely using a regex, `[a-z][a-zA-Z0-9_]*\.equals\((\"|[A-Z_]+\))` (intended to catch `javaVar.equals("string")` and `javaVar.equals(STRING_CONSTANT)`). You could add this regex to the checkstyle config: https://github.com/apache/incubator-druid/blob/master/codestyle/checkstyle.xml#L168

Reply:
Added.